frame/election-provider-multi-phase/src/signed.rs

```diff
@@ -43,6 +49,34 @@ pub struct SignedSubmission<AccountId, Balance: HasCompact, CompactSolution> {
 	pub solution: RawSolution<CompactSolution>,
 }
 
+impl<AccountId, Balance, CompactSolution> Ord
+	for SignedSubmission<AccountId, Balance, CompactSolution>
+where
+	AccountId: Ord,
+	Balance: Ord + HasCompact,
+	CompactSolution: Ord,
+{
+	fn cmp(&self, other: &Self) -> Ordering {
+		self.solution
+			.score
+			.cmp(&other.solution.score)
```

Comment:
this cmp is not taking `T::SolutionImprovementThreshold` into account.

Reply:
That's true, because 
1. `SignedSubmission` is not defined in terms of `T`, and 
2. We can't really produce a stable total ordering if we _do_ take the `SolutionImprovementThreshold` into account.

Reply:
That's informative, but not a solution :D 

1. You could make `SignedSubmission` have a `T: Get<Threshold>` and contain an additional `PhantomData<T>`
2. I can't see why. We have `ThresholdOrd` with the same API as `Ord`. An implementation of `Ord` can use `ThresholdOrd`


All in all, this is [uncool](https://www.youtube.com/watch?v=dy5S5kFZkzo); we can't throw away the entire use case of `SolutionImprovementThreshold` without having any second thought about it.

Reply:
Sorry, I was wrong: I am reading the rest of the code, and I realized that we do take into account later when there is a congestion and one solution needs to be ejected. I think that's enough.

Reply:
I didn't know about `ThresholdOrd`, which is cool, but it doesn't entirely solve the problem. According to the definition of `ThresholdOrd`, two items are equal if the greater is in the range of the lesser plus the threshold. `BTreeSet` is likely to get very confused if `Ord` says two items are equal but `Eq` disagrees. 

In any case, I'm not proposing to throw away `SolutionImprovementThreshold` entirely; there already exists some code checking it, and that could be extended to cover the full range of cases. It's just not in the `Ord` implementation.

https://github.com/paritytech/substrate/blob/33e6e456c55e4ae47c281ddb92e1a96e6ce85467/frame/election-provider-multi-phase/src/signed.rs#L217-L221

Reply:
My comment was immature, I should have read the code; all good 💯